Xamarin.PropertyEditing/ViewModels/ImageBrushViewModel.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Xamarin.PropertyEditing.Drawing;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Xamarin.PropertyEditing.ViewModels
+{
+	internal class ImageBrushViewModel : NotifyingObject
+	{
+		public ImageBrushViewModel (BrushPropertyViewModel parent)
+		{
+			Parent = parent ?? throw new ArgumentNullException (nameof (parent));
+			parent.PropertyChanged += Parent_PropertyChanged;
+		}
+
+		public CommonImageBrush PreviousImageBrush { get; set; }
+
+		public CommonTileMode TileMode {
+			get => Parent.Value is CommonImageBrush imageBrush ? imageBrush.TileMode : CommonTileMode.None;
+			set {
+				if (TileMode != value) {
+					Parent.Value = (Parent.Value as CommonImageBrush ?? new CommonImageBrush()).CopyWith (tileMode: value);
+					OnPropertyChanged ();
+				}
+			}
+		}
+
+		public IEnumerable<KeyValuePair<CommonTileMode, string>> TileModeValues
+			=> typeof (CommonTileMode).GetEnumValues ().Cast<CommonTileMode> ()
+				.Select (v => new KeyValuePair<CommonTileMode, string> (v, typeof (CommonTileMode).GetEnumName (v)))
+				.OrderBy (kvp => kvp.Key);
+
+		public CommonStretch Stretch
+		{
+			get => Parent.Value is CommonImageBrush imageBrush ? imageBrush.Stretch : CommonStretch.None;
+			set {
+				if (Stretch != value) {
+					Parent.Value = (Parent.Value as CommonImageBrush ?? new CommonImageBrush ()).CopyWith (stretch: value);
+					OnPropertyChanged ();
+				}
+			}
+		}
+
+		public IEnumerable<KeyValuePair<CommonStretch, string>> StretchValues
+			=> typeof (CommonStretch).GetEnumValues ().Cast<CommonStretch> ()
+				.Select (v => new KeyValuePair<CommonStretch, string> (v, typeof (CommonStretch).GetEnumName (v)))
+				.OrderBy (kvp => kvp.Key);
+
+		public CommonImageSource ImageSource
+		{
+			get => Parent.Value is CommonImageBrush imageBrush ? imageBrush.ImageSource : null;
+			set {
+				if (ImageSource != value) {
+					Parent.Value = (Parent.Value as CommonImageBrush ?? new CommonImageBrush ())
+						.CopyWith (imageSource: value);
+					OnPropertyChanged ();
+				}
+			}
+		}
+
+		public IReadOnlyList<KeyValuePair<CommonImageSource, string>> ImageSources
```

Comment:
I realize the assumption here is that the UI will query it once and that's all we need to worry about, but let's treat a property as it should be. We need an async value that we need to do some post-processing on, so let's fire the async request at `ctor` and do the transform task-inline (`ContinueWith` / `ConfigureAwait (false)`.) We should not re-query this end result, it should be cached. In the property we _could_ then block on this final cached task, but it's entirely possible this query time will be significant. There's a type in the library to deal with this exact scenario (`AsyncValue`) to which you can hand a task and bind to the `.Value` of it and have the binding system automatically update when it's finished loading. Additionally it provides a loading indicator that can be bound to.